Fix typing in _app_model

dstansby opened the pull request:
Fixes most typing in `napari._app_model`. The main change is renaming `CommandId.title` to `CommandId.command_title` to avoid overriding the `.title` property of the superclass (`str`).

dstansby commented on napari/_app_model/constants/_commands.py:
```diff
@@ -75,7 +75,6 @@ class CommandId(str, Enum):
     LAYER_PROJECT_MEAN = 'napari:layer:project_mean'
     LAYER_PROJECT_MEDIAN = 'napari:layer:project_median'
 
-    @property
```
This change has been made to make `title()` a method, consistent with `string` which is this classes supertype.

Since this is in an underscored file I assume it's private API and fine to change without deprecation?

aganders3 replied:
This is a little confusing to me as I feel `title()` on `str` does something different (returns a title-cased version of the string) from what is intended here (returning the "title" of the command). The name collision is unfortunate though.

Is it necessary for this class to inherit from `str`? Maybe it could use [StringEnum](https://github.com/napari/napari/blob/9bea15e7b89156654ae02c3dc3144d6fc244aab3/napari/utils/misc.py#L314) instead?

dstansby replied:
How about renaming the property to `.command_title` here?

Carreau replied:
I'm pretty sure there already is a discussion on this topic somewhere as I already encountered this issue.

I believe the answer was that we can't change as it's public API and might be used in npe2/napari hub ?

If change there is, i believe we'll need proper deprecation, i.e. add a `.command_title`, and emit a deprecation warning for some period of time.

tlambert03 replied:
I'm not at my computer, but if I recall correctly this was strictly internal. Just a convenience.  So I would change the name if I were yoh

tlambert03 replied:
Almost nothing about naparis usage of app model should be public api, unless that has changed since I worked on it

Carreau replied:
Thanks ! My memory failed me then :-) 

+1 to rename then.

dstansby replied:
👍 I changed the name, was easier to redo the commit so I force pushed away the original commit.

dstansby committed:
Fix typing in _app_model

Fixes most typing in `napari._app_model`. The main change is renaming
`CommandId.title` to `CommandId.command_title` to avoid overriding the
`.title` property of the superclass (`str`).

dstansby committed:
Merge branch 'main' into typing-_app-model

dstansby committed:
Merge branch 'main' into typing-_app-model

dstansby committed:
Fix command_title in _file
